src/lib/shared/form/side_effects.ts

```diff
@@ -34,7 +34,8 @@ export const getFormEffects = <T extends Values>(
     action$.pipe(
       filter(() => typeof propsRef.current?.onSubmit === 'function'),
       isActionOf(actions.Submit),
-      switchMap(([_, __, { values }]) => {
+      switchMap(([_, __, { values, isValid }]) => {
+        if (!isValid) return of()
```

Comment:
Empty makes more sense here
```suggestion
        if (!isValid) return empty()
```